Server/src/main/java/org/gluu/oxauth/model/token/IdTokenFactory.java

```diff
@@ -274,9 +275,12 @@ public Jwt generateSignedIdToken(IAuthorizationGrant authorizationGrant, String
             if (authorizationGrant.getClient().getSubjectType() != null && SubjectType.fromString(authorizationGrant.getClient().getSubjectType()).equals(SubjectType.PAIRWISE)) {
                 log.warn("Unable to calculate the pairwise subject identifier because the client hasn't a redirect uri. A public subject identifier will be used instead.");
             }
-
             String openidSubAttribute = appConfiguration.getOpenidSubAttribute();
-            jwt.getClaims().setSubjectIdentifier(authorizationGrant.getUser().getAttribute(openidSubAttribute));
+            String subValue = authorizationGrant.getUser().getAttribute(openidSubAttribute);
+            if (StringHelper.equalsIgnoreCase(openidSubAttribute, "uid")) {
+                subValue = authorizationGrant.getUser().getUserId();
+            }
+            jwt.getClaims().setSubjectIdentifier(subValue);
```

Comment:
@miltonbo please check `org.gluu.oxauth.model.common.AuthorizationGrant#getSub`. I guess that method has to be corrected in latest master.